### PR TITLE
feat: add base promotional advert

### PR DESCRIPTION
**Pull request by JFrankfurt**
https://uniswapteam.slack.com/archives/C047U65H422/p1691590740068649

image
image

### CODE REVIEW

File: src/components/Banner/BaseAnnouncementBanner/index.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  const shouldDisplay = true || Boolean(!hideUniswapWalletBanner && !isLandingScreen)
+
+  const screenSize = useScreenSize()
+
+  if (isMobileSafari) return null
+
+  return (
+    <PopupContainer show={shouldDisplay}>
+      <StyledXButton
```
Comment (cbachmeier) [inline src/components/Banner/BaseAnnouncementBanner/index.tsx:27]: Advert does not close when `X` button is clicked
advertClose

File: src/components/TopLevelModals/index.tsx [hunk 2]
```diff
@@ -28,7 +28,8 @@ export default function TopLevelModals() {
       <ConnectedAccountBlocked account={account} isOpen={accountBlocked} />
       <Bag />
       <UniwalletModal />
-      <UniswapWalletBanner />
+      <BaseAnnouncementBanner />
+      {/* <UniswapWalletBanner /> */}
```
Comment (cbachmeier) [inline src/components/TopLevelModals/index.tsx:32]: Can we delete the old `components/Banner/UniswapWalletBanner` file?

File: src/components/Banner/BaseAnnouncementBanner/styled.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  @media only screen and (max-width: ${({ theme }) => `${theme.breakpoint.md}px`}) {
+    bottom: 62px;
+  }
+
+  @media only screen and (max-width: ${({ theme }) => `${theme.breakpoint.sm}px`}) {
+    background-position: top 32px right -10px;
+    width: unset;
+    right: 10px;
+    left: 10px;
+    height: 144px;
```
Comment (cbachmeier) [inline src/components/Banner/BaseAnnouncementBanner/styled.tsx:53]: this wasn't in the UniswapWalletBanner `PopupContainer` style, making sure this is intentional?

File: src/components/Banner/BaseAnnouncementBanner/index.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        size={20}
+        onClick={(e) => {
+          // prevent click from bubbling to UI on the page underneath, i.e. clicking a token row
+          e.preventDefault()
+          e.stopPropagation()
+          toggleHideUniswapWalletBanner()
+        }}
+      />
+
+      <BaseBackgroundImage src={baseLogoUrl} alt="" />
```
Comment (cbachmeier) [inline src/components/Banner/BaseAnnouncementBanner/index.tsx:38]: nit: add an alt text description of the abnner

File: src/components/Banner/BaseAnnouncementBanner/index.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+              <ThemedText.LabelSmall color="black" marginLeft="5px">
+                {!screenSize['xs'] ? <Trans>Download</Trans> : <Trans>Download app</Trans>}
+              </ThemedText.LabelSmall>
+            </BannerButton>
+
+            <BannerButton backgroundColor="black" onClick={() => openWalletMicrosite()}>
+              <ThemedText.LabelSmall color="white">
+                <Trans>Learn more</Trans>
+              </ThemedText.LabelSmall>
+            </BannerButton>
```
Comment (cbachmeier) [inline src/components/Banner/BaseAnnouncementBanner/index.tsx:77]: nit: don't duplicate this component and instead conditionally set its coloring based on `isIOS`

**Review by cbachmeier — CHANGES_REQUESTED** [review]

File: src/components/Banner/BaseAnnouncementBanner/styled.tsx (hunk 3 above)
Comment (JFrankfurt) [reply src/components/Banner/BaseAnnouncementBanner/styled.tsx:53]: yeah, this is a new addition to this component

File: src/components/TopLevelModals/index.tsx (hunk 2 above)
Comment (JFrankfurt) [reply src/components/TopLevelModals/index.tsx]: No, the new one is just a re-skin for Base specifically. It's only rendered on that network

File: src/components/Banner/BaseAnnouncementBanner/index.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+import { useLocation } from 'react-router-dom'
+import { useHideUniswapWalletBanner } from 'state/user/hooks'
+import { ThemedText } from 'theme'
+import { openDownloadApp, openWalletMicrosite } from 'utils/openDownloadApp'
+import { isIOS, isMobileSafari } from 'utils/userAgent'
+
+import { BannerButton, BaseBackgroundImage, ButtonRow, PopupContainer, StyledXButton } from './styled'
+
+export default function UniswapWalletBanner() {
+  const [hideUniswapWalletBanner, toggleHideUniswapWalletBanner] = useHideUniswapWalletBanner()
```
Comment (cbachmeier) [inline src/components/Banner/BaseAnnouncementBanner/index.tsx:15]: Do we want to reset the state for this to pump Base launch? For example I was only able to see this banner in incognito

**Review by natew — APPROVED** [review]
beyond charlies comments lgtm

File: src/state/user/reducer.test.ts [hunk 7]
```diff
@@ -110,10 +110,10 @@ describe('swap reducer', () => {
     })
   })
 
-  describe('updateHideUniswapWalletBanner', () => {
+  describe('updateHideBaseWalletBanner', () => {
     it('updates the hideUniswapWalletBanner', () => {
```
Comment (cbachmeier) [inline src/state/user/reducer.test.ts:114]: update test description text

File: src/components/Banner/BaseAnnouncementBanner/index.tsx [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+          Swap on{' '}
+          <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <path
+              fillRule="evenodd"
+              clipRule="evenodd"
+              d="M19.5689 10C19.5689 15.4038 15.1806 19.7845 9.76737 19.7845C4.63163 19.7845 0.418433 15.8414 0 10.8225H12.9554V9.17755H0C0.418433 4.15863 4.63163 0.215576 9.76737 0.215576C15.1806 0.215576 19.5689 4.59621 19.5689 10Z"
+              fill="white"
+            />
+          </svg>{' '}
+          BASE in the Uniswap wallet
```
Comment (cbachmeier) [inline src/components/Banner/BaseAnnouncementBanner/index.tsx:54]: not blocking but for design: <360px and white text can't be see on white phone image cc: @infredible 
Screenshot 2023-08-09 at 3 50 40 PM

File: src/components/Banner/BaseAnnouncementBanner/index.tsx [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+            </BannerButton>
+
+            <BannerButton backgroundColor="black" onClick={() => openWalletMicrosite()}>
+              <ThemedText.LabelSmall color="white">
+                <Trans>Learn more</Trans>
+              </ThemedText.LabelSmall>
+            </BannerButton>
+          </>
+        ) : (
+          <BannerButton backgroundColor="white" width="125px" onClick={() => openWalletMicrosite()}>
```
Comment (cbachmeier) [inline src/components/Banner/BaseAnnouncementBanner/index.tsx:83]: nit 
```suggestion
          <BannerButton backgroundColor="white" width="125px" onClick={openWalletMicrosite}>
```

File: src/components/Banner/BaseAnnouncementBanner/index.tsx (hunk 8 above)
Comment (JFrankfurt) [reply src/components/Banner/BaseAnnouncementBanner/index.tsx:54]: iphone mini is 375 and looks alright, so I think this is fine?

image

File: src/components/Banner/BaseAnnouncementBanner/index.tsx (hunk 9 above)
Comment (JFrankfurt) [reply src/components/Banner/BaseAnnouncementBanner/index.tsx]: ts doesn't like this since the handler arguments don't agree with the action args

**Review by cbachmeier — APPROVED** [review]

**Pull request merged**